### PR TITLE
Add Tooltips

### PR DESCRIPTION
**Pull request by leha-code**
Added the tooltip code and tooltips

### CODE REVIEW

**Comment by leha-code** [comment]
I did test the code

**Comment by leha-code** [comment]
also changed the cursor to a rocket on the launch button

**Comment by leha-code** [comment]
also hey #git-webhooks

**Comment by leha-code** [comment]
also I fixed a typo where there was pixmap instead of pixmaps

**Comment by donnerkebab1730** [comment]
i would say your good to go except i cant read code well so i guess its someone elses job

**Comment by leha-code** [comment]
The code is from an online tutorial, yes

File: src/mcpil.py [hunk 1]
```diff
@@ -328,6 +357,8 @@ def play_tab(parent):
             title = ttk.Label(tab, text='Minceraft Pi Launcher')
         else:
             title = ttk.Label(tab, text='Minecraft Pi Launcher')
+
+    title.tooltip = ToolTip(title, text="This is the title, what are you gazing at?\nThere's a small chance it will show something else than this plain title...")
```
Comment (nobody5050) [inline src/mcpil.py:361]: Unnecessary imo

File: src/mcpil.py [hunk 2]
```diff
@@ -346,6 +377,7 @@ def play_tab(parent):
 
     splash_text = ttk.Label(tab, text=splash, foreground='yellow')
     splash_text.grid(row=1, pady=4)
+    splash_text.tooltip = ToolTip(splash_text, text="This is the random splash that is\nsometimes very interesting...\nFor example at a special day...")
```
Comment (nobody5050) [inline src/mcpil.py:380]: Unnecessary as well

File: src/mcpil.py (hunk 1 above)
Comment (leha-code) [reply src/mcpil.py:361]: Come on, don't you like jokes?

Comment (mobilegmYT) [reply src/mcpil.py:361]: It's the easter eggs that make things interesting :)

File: src/mcpil.py [hunk 3]
```diff
@@ -439,6 +479,7 @@ def features_tab(parent):
     row = 0
     for key in launcher.AVAILABLE_FEATURES:
         check = ttk.Checkbutton(main_frame.scrollable_frame, command=update_features, text=key)
+        check.tooltip = ToolTip(check, text=f"Check this if you want to {key} to be enabled.")
```
Comment (nobody5050) [inline src/mcpil.py:482]: Maybe change to 
```python

check.tooltip = ToolTip(check, text=f"Check this if you want {key} to be enabled.")

```

File: src/mcpil.py [hunk 4]
```diff
@@ -538,7 +583,7 @@ def main():
     window.resizable(True, True)
 
     # Set icon in taskbar
-    window.iconphoto(True, PhotoImage(file="/usr/share/pixmap/mcpil.png"))
+    window.iconphoto(True, PhotoImage(file="/usr/share/pixmaps/mcpil.png"))
```
Comment (nobody5050) [inline src/mcpil.py:586]: What does this change?

**Review by nobody5050 — CHANGES_REQUESTED** [review]
Maybe change some things before merge

File: src/mcpil.py (hunk 1 above)
Comment (nobody5050) [reply src/mcpil.py:361]: Yeah but it’s not really an Easter egg

Comment (nobody5050) [reply src/mcpil.py:361]: Also, new lines in tooltips is ugly. They should be short, not a whole paragraph

File: src/mcpil.py (hunk 4 above)
Comment (leha-code) [reply src/mcpil.py:586]: Ooh, this makes the image actually appear without errors, rpinews or alv mistyped

File: src/mcpil.py [hunk 5]
```diff
@@ -470,19 +512,22 @@ def multiplayer_tab(parent):
     current_ip = StringVar(main_frame)
     current_ip.trace('w', lambda *args: update_proxy())
     ip = ttk.Entry(main_frame, width=24, textvariable=current_ip)
+    ip.tooltip = ToolTip(ip, text="This is the IP of the multiplayer server.\nThe IP might be a domain, for example, thebrokenrail.com,\nor it might be an IP adress such as 192.168.1.3")
     ip.grid(row=0, column=1, padx=6, pady=6, sticky='EW')
```
Comment (leha-code) [inline src/mcpil.py:517]: Oh,right, I'm on an eBook and as an admin you can change stuff in my fork, so if you want to change stuff, change!

Comment (nobody5050) [reply src/mcpil.py:517]: ?

Comment (nobody5050) [reply src/mcpil.py:517]: I didn’t request changes to this

**Comment by leha-code** [comment]
There

**Review by nobody5050 — APPROVED** [review]

File: src/mcpil.py (hunk 4 above)
Comment (Alvarito050506) [reply src/mcpil.py:584]: This is correct, probably RPiNews or me made a typo.

Comment (leha-code) [reply src/mcpil.py:584]: Lol

**Comment by TheBrokenRail** [comment]
I disagree with this for the same reason disagreed with MCPI-Revival/MCPIedit#2:

> Ah yes, the world name text box is made of world name. What benefit does this add apart from cluttering the interface?

All of these text boxes and other miscellaneous widgets already have clearly visible labels. What does this add besides code complexity?

However, this has already been merged and this project will probably be deprecated in favor of gMCPIL or jMCPIL eventually, so I'm going to leave this in-place.

**Pull request merged**